Core/Graphics/Source/FrameBuffer.cpp

```diff
@@ -53,37 +53,72 @@ namespace Babylon::Graphics
 
     void FrameBuffer::Clear(bgfx::Encoder& encoder, uint16_t flags, uint32_t rgba, float depth, uint8_t stencil)
     {
-        // BGFX requires us to create a new viewID, this will ensure that the view gets cleaned.
+        // BGFX requires us to create a new viewID, this will ensure that the view gets cleared.
         m_viewId = m_context.AcquireNewViewId(encoder);
 
         bgfx::setViewMode(m_viewId.value(), bgfx::ViewMode::Sequential);
         bgfx::setViewClear(m_viewId.value(), flags, rgba, depth, stencil);
         bgfx::setViewFrameBuffer(m_viewId.value(), m_handle);
 
-        // BGFX will consider the viewport when cleaning the screen, but WebGL always cleans the entire screen.
-        // That's why we always set the viewport to {0, 0, 1, 1} when cleaning.
-        bgfx::setViewRect(m_viewId.value(), 0, 0, Width(), Height());
-        encoder.touch(m_viewId.value());
+        // If a scissor is not set, WebGL clears the entire screen, so set the view rect to cover the entire screen
+        // before clearing to match WebGL's behavior; otherwise BGFX will only clear the view rect.
+        //
+        // If a scissor is set, we need to set the BGFX view rect to match it before clearing.
+        // We set the view rect instead of the view scissor because BGFX clears after the view rect is set and before
+        // the view scissor is set.
+        //
+        // Note that the view rect and view scissor are reset to the desired dimensions before the encoder is submitted.
+        if (m_desiredScissor.X == 0.0f && m_desiredScissor.Y == 0.0f && m_desiredScissor.Width == 0.0f && m_desiredScissor.Height == 0.0f)
```

Comment:
float comparison is a bit dangerous. use a fabsf with an epsylon instead. This said, x,y,width and height are in pixels, so why not keep uint16 for these values all along? And if needed, use floats at the very end.

Reply:
These values are all explicitly set to 0.0 when the scissor is disabled, so I don't think an epsilon is needed here.